<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TSP_SolverWindow</class>
 <widget class="QWidget" name="TSP_SolverWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>808</width>
    <height>701</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <property name="spacing">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QWidget" name="centralWidget" native="true">
     <layout class="QVBoxLayout" name="centralWidget_2">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>1</number>
      </property>
      <property name="topMargin">
       <number>1</number>
      </property>
      <property name="rightMargin">
       <number>1</number>
      </property>
      <property name="bottomMargin">
       <number>1</number>
      </property>
      <item>
       <widget class="QWidget" name="widget_2" native="true">
        <layout class="QGridLayout" name="gridLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <property name="horizontalSpacing">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget" native="true">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>200</height>
         </size>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <property name="spacing">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QTabWidget" name="tabWidget">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="tabGA">
            <attribute name="title">
             <string>Генетический алгоритм</string>
            </attribute>
            <widget class="QPushButton" name="StartGABut">
             <property name="geometry">
              <rect>
               <x>30</x>
               <y>50</y>
               <width>121</width>
               <height>41</height>
              </rect>
             </property>
             <property name="text">
              <string>Старт</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
            <widget class="QPushButton" name="StopGABut">
             <property name="geometry">
              <rect>
               <x>30</x>
               <y>100</y>
               <width>121</width>
               <height>41</height>
              </rect>
             </property>
             <property name="text">
              <string>Стоп</string>
             </property>
             <property name="checkable">
              <bool>false</bool>
             </property>
            </widget>
            <widget class="QSpinBox" name="spinBox">
             <property name="geometry">
              <rect>
               <x>400</x>
               <y>60</y>
               <width>61</width>
               <height>31</height>
              </rect>
             </property>
             <property name="maximum">
              <number>100000</number>
             </property>
             <property name="value">
              <number>5000</number>
             </property>
            </widget>
            <widget class="QSpinBox" name="spinBox_2">
             <property name="geometry">
              <rect>
               <x>400</x>
               <y>100</y>
               <width>61</width>
               <height>31</height>
              </rect>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
            </widget>
            <widget class="QSpinBox" name="spinBox_3">
             <property name="geometry">
              <rect>
               <x>560</x>
               <y>40</y>
               <width>51</width>
               <height>31</height>
              </rect>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>20</number>
             </property>
            </widget>
            <widget class="QSpinBox" name="spinBox_4">
             <property name="geometry">
              <rect>
               <x>560</x>
               <y>80</y>
               <width>51</width>
               <height>31</height>
              </rect>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>55</number>
             </property>
            </widget>
            <widget class="QSpinBox" name="spinBox_5">
             <property name="geometry">
              <rect>
               <x>560</x>
               <y>120</y>
               <width>51</width>
               <height>31</height>
              </rect>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>75</number>
             </property>
            </widget>
            <widget class="QLabel" name="label">
             <property name="geometry">
              <rect>
               <x>300</x>
               <y>60</y>
               <width>91</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Размер популяции</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_2">
             <property name="geometry">
              <rect>
               <x>300</x>
               <y>100</y>
               <width>91</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Элитарность</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_3">
             <property name="geometry">
              <rect>
               <x>480</x>
               <y>40</y>
               <width>71</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Мутация 1</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_4">
             <property name="geometry">
              <rect>
               <x>480</x>
               <y>80</y>
               <width>71</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Мутация 2</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_5">
             <property name="geometry">
              <rect>
               <x>480</x>
               <y>120</y>
               <width>71</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Мутация 3</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_6">
             <property name="geometry">
              <rect>
               <x>30</x>
               <y>20</y>
               <width>111</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>Работа Алгоритма</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_7">
             <property name="geometry">
              <rect>
               <x>300</x>
               <y>20</y>
               <width>121</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>Параметры Алгоритма</string>
             </property>
            </widget>
           </widget>
           <widget class="QWidget" name="tabBB">
            <attribute name="title">
             <string>Метод ветвей и границ</string>
            </attribute>
            <widget class="QPushButton" name="StartBBBut">
             <property name="geometry">
              <rect>
               <x>30</x>
               <y>50</y>
               <width>121</width>
               <height>41</height>
              </rect>
             </property>
             <property name="text">
              <string>Старт</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
            <widget class="QPushButton" name="StopBBBut">
             <property name="geometry">
              <rect>
               <x>30</x>
               <y>100</y>
               <width>121</width>
               <height>41</height>
              </rect>
             </property>
             <property name="text">
              <string>Стоп</string>
             </property>
             <property name="checkable">
              <bool>false</bool>
             </property>
            </widget>
            <widget class="QLabel" name="label_8">
             <property name="geometry">
              <rect>
               <x>30</x>
               <y>20</y>
               <width>111</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>Работа Алгоритма</string>
             </property>
            </widget>
           </widget>
           <widget class="QWidget" name="tabWM">
            <attribute name="title">
             <string>Матрица весов</string>
            </attribute>
            <layout class="QGridLayout" name="gridLayout_3">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <property name="spacing">
              <number>0</number>
             </property>
             <item row="0" column="0">
              <widget class="QTextEdit" name="textEdit">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="UpdateBut">
               <property name="text">
                <string>Обновить матрицу</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
  <action name="actionStartGA">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>StartGA</string>
   </property>
  </action>
  <action name="actionStartBB">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>StartBB</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Очистить граф</string>
   </property>
  </action>
  <action name="actionStopGA">
   <property name="text">
    <string>StopGA</string>
   </property>
   <property name="toolTip">
    <string>StopGA</string>
   </property>
  </action>
  <action name="actionStopBB">
   <property name="text">
    <string>StopBB</string>
   </property>
   <property name="toolTip">
    <string>StopBB</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>StartGABut</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionStartGA</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>419</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StopBBBut</sender>
   <signal>pressed()</signal>
   <receiver>actionStopBB</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>289</x>
     <y>419</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StopGABut</sender>
   <signal>pressed()</signal>
   <receiver>actionStopGA</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>77</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StartBBBut</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionStartBB</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>92</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
